lib/mongoose_push/application.ex

```diff
@@ -6,22 +6,37 @@ defmodule MongoosePush.Application do
   use Application
   require Logger
 
+  @typedoc "Possible keys in both services"
+  @type fcm_key :: :key | :pool_size | :mode | :endpoint
+  @type apns_key :: :cert | :key | :pool_size | :mode | :endpoint | :use_2197
+
+  @typedoc """
+  In FCM `:key` and `:pool_size` are required and `:mode` has to be either `:dev` or `:prod`
+  """
+  @type fcm_config :: [fcm_key: String.t() | atom | integer]
+  @typedoc """
+  In APNS `:cert`, `:key` and `:pool_size` are required. `:mode` has to be either `:dev` or `:prod`
+  """
+  @type apns_config :: [apns_key: String.t() | atom | integer]
```

Comment:
According to docs this is not the thing you probably intended. This means, that we have a keyword list with a key `apns_key` of type `String.t() | atom | integer`.
I think that it needs to be:
`[{apns_key, String.t() | atom | integer}]`